packages/dev/core/src/Engines/WebGL/webGLPipelineContext.ts

```diff
@@ -110,32 +170,45 @@ export class WebGLPipelineContext implements IPipelineContext {
     }
 
     /**
-     * @param uniformName
-     * @param x
-     * @param y
+     * @param _uniformName
+     * @param _x
+     * @param _y
+     * @param _z
+     * @param _w
      * @hidden
      */
-    public _cacheFloat2(uniformName: string, x: number, y: number): boolean {
-        let cache = this._valueCache[uniformName];
+    public _cacheFloatN(_uniformName: string, _x: number, _y: number, _z?: number, _w?: number): boolean {
+        /**
+         * arguments will be used to abstract the cache function.
+         * arguments[0] is the uniform name. the rest are numbers.
+         */
+        let cache: number[] = this._valueCache[arguments[0]];
         if (!cache || cache.length !== 2) {
-            cache = [x, y];
-            this._valueCache[uniformName] = cache;
+            cache = Array.prototype.slice.call(arguments, 1);
+            this._valueCache[arguments[0]] = cache;
             return true;
         }
 
         let changed = false;
-        if (cache[0] !== x) {
-            cache[0] = x;
-            changed = true;
-        }
-        if (cache[1] !== y) {
-            cache[1] = y;
-            changed = true;
-        }
-
+        cache.forEach(function (num, idx) {

```

Comment:
is this not GCing the function on every call ?

Reply:
good catch, changing to for